---
title: "Dissecting Photoreceptor Contributions to Leaf Growth"
subtitle: "Dantony Lewis"
format:
  revealjs:
    incremental: true   
    theme: simple
---

## Photoreceptors

- Plants possess several photoreceptor proteins that percieve light from blue to far red

- Plants use light signals to activate photoreceptors that interact with transcription factors (PIFs) in order to impact hormone signaling pathways that adjust cellular processes

::: {.slide style="text-align: center;"}
![](images/visible-light-spectrum-infared-and-ultraviolet-electromagnetic-visible-color-spectrum-for-human-eye-gradient-diagram-with-wavelength-and-colors-educational-illustration-on-white-background-vector.jpg){width=70% height=70%}
:::

## Phytochrome & Cryptochrome

- Phytochrome & Cyrptochrome are the essential photoreceptor proteins but respond to different wavelengths of light 

- The main types of Photoreceptors that signal different functions throughout plant life are: *phyA*, *phyB*, *cry1*, *cry2*

## Question
Which photoreceptor (*phyA*, *phyB*, *cry1*, *cry2*) promotes leaf growth in the light responsive gene network of the *Arabidopsis thaliana*. 

**Prior Studies**

- *phyB* red light signaling inhibits the initiation of leaf senescence^[(Sakuraba, 2021)]

- Other photoreceptors (*cry1* & *phyA*) inhibit *phyB* suppression of floral initiation^[(Mockler, 1999)]

::: aside

:::

## Hypothesis

- *phyB* significantly promotes active light responses in leaf growth within the light responsive gene network of *A. thaliana*

::: {.slide style="text-align: center;"}
![](images/Arabidopsis_thaliana_rosette_transparent_background.png){width=40% height=40%}
:::

## Data {.smaller}
Collected by: Department of Biotechnology and Life Sciences, University of Insubria, Varese (VA), Italy

- *A. thaliana* mutants containing the loss-of-function in specific photoreceptors grown under high-pressure sodium (HPS) lamps

- Lamina / Petiole ratio (L/P) measured when the six-leaf stage is reached ~17 days after sowing(DAS)

::: {.slide style="text-align: center;"}
![](images/leaf_diagram.jpg){width=30% height=30%}
:::

## Base Data Visualization
```{r}
#| warning: false
#| message: false
library(tidyverse)
base_d <- read_csv("data/revised_data.csv")
# Base Viz
base_viz <- ggplot(data = base_d,
                   aes(x = Mutant,
                       y = `L/P`,
                       color = Mutant)) +
  scale_color_viridis_d() +
  geom_boxplot()
base_viz
```

## Linear Model
$$
L/P_i = \beta_0 + \sum_{j=1}^{k} \beta_j \cdot \text{Mutant}_{ij} + \varepsilon_i
$$


::::: columns
::: {.column width="60%"}
```{r}
#| warning: false
#| message: false
# Base Model 
library(performance)
library(broom)
library(car)

base_d <- base_d |>
  mutate(Mutant = factor(Mutant, levels = c("WT", setdiff(unique(Mutant), "WT"))))


lm_model <-  lm(log(`L/P`) ~ Mutant, data = base_d)

check_residuals(lm_model) |> plot()
```
:::
::: {.column width="40%"}
```{r}

tidy(lm_model)[, c("term", "p.value")] |>
  select(Photoreceptor = term, p.value)
```
:::
::::
## Tukey's Test {.smaller}
Used to preform pairwise comparisons between different levels of factor in a linear model to determine significant difference.
```{r}
Anova_mod <- aov(lm_model)

Tukey_mod <- TukeyHSD(x = Anova_mod, "Mutant", conf.level = 0.95)

my_colors <- c("navy", "red", "orange", "navy", "black", "black", "black", "black", "black", "black")
par(mar=c(5,6,4,1)+.1)

Tukey_plot <- plot(Tukey_mod, las=1, col=my_colors)
```

## AIC Model Average {.smaller}
Combining the prediction of two models: null model (mean only) and categorical model (linear model), then reflecting the weighted average  
```{r}
#| warning: false
#| message: false
#| echo: false
#| results: hide
library(AICcmodavg)
library(MuMIn)

phyto_lm <- lm(log(`L/P`) ~ Mutant, data = base_d)
phyto_mean <- lm(log(`L/P`) ~ 1, data = base_d)

aictab(list(phyto_lm, phyto_mean), 
       c("linear model", "mean only"),
       second.ord = FALSE)

mod_list <- list(phyto_mean, phyto_lm)
mod_names <- c("mean only", "linear model")

mod_names <- 0:4

avg_model <- model.avg(mod_list)


new_data <- data.frame(Mutant = unique(base_d$Mutant))
preds <- predict(avg_model, newdata = new_data, se.fit = TRUE)

pred_df <- data.frame(
  Mutant = new_data$Mutant,
  fit = preds$fit,
  lower = preds$fit - 1.96 * preds$se.fit,
  upper = preds$fit + 1.96 * preds$se.fit)

aic_plot <- ggplot(pred_df, aes(x = Mutant, y = fit, color = "Predicted")) +
  geom_point(size = 3) +
  geom_errorbar(aes(ymin = lower, ymax = upper, color = "Predicted"), width = 0.2) +
  geom_jitter(data = base_d, aes(x = Mutant, y = `L/P`, color = "Observed"), 
              width = 0.05, alpha = 0.5, size = 2) +
  scale_color_manual(name = "Data Type",
                     values = c("Observed" = "skyblue", "Predicted" = "grey40")) +
   labs(y = "L/P Ratio", x = "Mutant", 
       title = "Observed and Model-Averaged Predicted L/P Values") +
  theme_minimal()
aic_plot
```
## Gene Expression Analyisis 
Data Collected by: Shanghai University, Shanghai, China

- Analyzing the gene expression profile of red-light grown *phyB* mutants by sequencing mRNA from samples to analyze the genes regulated by *phyB.* 

- Using limma an r package from Bioconductor, to analyze gene expression data and identity significant genes

::: {.slide style="text-align: center;"}
![](images/0*1EvyFTQv-JAZh5nM.png){width=17% height=17%}
:::

## limma Visualization {.smaller}
M value = difference between the log~2~-transformed expression values  
A value = average log~2~ expression
```{r}
#| warning: false
#| message: false
#| echo: false
#| results: hide

library(limma)

exprs <- read.csv("data/expression_matrix.csv", header=TRUE, row.names=1)
exprs <- exprs[, c("RPKM_PhyB", "RPKM_WT")]
colnames(exprs) <- c("PhyB", "WT")
exprs <- as.matrix(exprs)

metadata <- data.frame(
  Sample = c("PhyB", "WT"),
  Genotype = c("PhyB", "WT"),
  Ratio = c(NA, NA))  # no phenotype data
rownames(metadata) <- metadata$Sample
metadata$Genotype <- factor(metadata$Genotype)
exprs <- log2(exprs + 1)

design <- model.matrix(~ 0 + Genotype, data=metadata)
colnames(design) <- levels(metadata$Genotype)

fit <- lmFit(exprs, design)
contrast.matrix <- makeContrasts(PhyB_vs_WT = PhyB - WT, levels=design)
fit2 <- contrasts.fit(fit, contrast.matrix)

results <- fit2$coefficients

log2fc <- log2(exprs[, "PhyB"]) - log2(exprs[, "WT"])
results <- data.frame(
  Gene = rownames(exprs),
  log2FoldChange = log2fc
)
head(results[order(-abs(results$log2FoldChange)), ])

A <- (log2(exprs[, "PhyB"]) + log2(exprs[, "WT"])) / 2
M <- log2(exprs[, "PhyB"]) - log2(exprs[, "WT"])
limma_plot <- plot(A, M, pch=16, cex=0.6, main="MA Plot of PhyB constrasted to WT", 
     xlab="Average log2 Expression (A)", ylab="log2 Fold Change (M)",
     col=ifelse(abs(M) > 1, "gray", "black"))
abline(h=0, col="red")
highlight_genes <- abs(M) > 2
points(A[highlight_genes], M[highlight_genes], col="skyblue", pchisq=16, cex=1.2)

limma_plot
```
::: footer
[Blue]{style="color:skyblue;"} circles represent genes that are significantly differently expressed
:::
## Analysis
**Categorical Linear Model**   

- Mutant *phyB* impact on leaf growth (L/P) is significantly different from the WT  

- *phyA* & *cry2* show no clear effect on leaf growth, while *cry1* suggests some difference  

**limma MA plot**  

- There are significant genes that likely play a role in how *phyB* differs biologically from the WT  

## Further Exploration
 - Is leaf growth (L/P) predictive of *phyB* gene expression patterns in *Arabidopsis thaliana*? 
 - Measure a phenotypic factor as well as gene expression to create a model that utilizes both
 - With use of significant *phyB* genes, we can find which genes' expression level change with leaf growth and if *phyB*-regulated genes are among these
 
## References {.scrollable .smaller}

::: {.nonincremental}
- Beatrice, Peter, et al. “Photoreceptors’ gene expression of Arabidopsis thaliana grown with biophilic LED-sourced lighting systems.” PLOS ONE, vol. 17, no. 6, 10 June 2022

- Boccalandro, Hernán E., et al. “Phytochrome B enhances photosynthesis at the expense of water-use efficiency in Arabidopsis.” Plant Physiology, vol. 150, no. 2, 10 Apr. 2009, pp. 1083–1092

- Colón-Carmona, Adán (2025, March). Lecture 12: Light Signaling & Responses [Lecture]. BIOL 321: Plant Physiology, University of Massachusetts Boston

- Kozuka, Toshiaki, et al. “The different growth responses of the Arabidopsis thaliana leaf blade and the petiole during shade avoidance are regulated by photoreceptors and sugar.” Plant and Cell Physiology, vol. 46, no. 1, 15 Jan. 2005, pp. 213–223

- Mockler, Todd C., et al. “Antagonistic actions of arabidopsis cryptochromes and phytochrome B in the regulation of Floral Induction.” Development, vol. 126, no. 10, 15 May 1999, pp. 2073–2082

- Sakuraba, Yasuhito. “Light-mediated regulation of leaf senescence.” International Journal of Molecular Sciences, vol. 22, no. 7, 24 Mar. 2021, p. 3291
:::


